### PR TITLE
"archive" becomes "withdrawn" in admin interface and codebase

### PR DESCRIPTION
**Pull request by tekin**
This is a transitional set of changes that retitles the concept of "archiving" an edition to "withdrawing", matching up with the new [retention policy guidandance](https://www.gov.uk/guidance/content-design/gov-uk-content-retention-and-withdrawal-archiving-policy).

## Before

screen shot 2015-05-28 at 11 29 18

screen shot 2015-05-28 at 11 29 14

screen shot 2015-05-28 at 11 29 20

screen shot 2015-05-28 at 11 29 24

## After

screen shot 2015-05-28 at 11 28 36

screen shot 2015-05-28 at 11 28 38

screen shot 2015-05-28 at 11 28 34

screen shot 2015-05-28 at 11 28 31

As well as updating the UI in admin to use the term "withdraw" instead of "archive", it also updates the codebase to use the new terminology. Existing archived editions are left in an `"archived"`state, and editions that are withdrawn in the future will transition to a `"withdrawn"` state. The changes have been made such that the app will treat editions in an `"archived"` and `"withdrawn"` state as the same. Once these changes have been successfully deployed, we can run a data migration to change `"archived"` editions to `"withdrawn"` and update the codebase to get rid of any transitional code in place to handle the dual-states.

## Post-deploy changes to complete transition

* Data migration to set "archived" edition states to "withdrawn"
* Replace temporary `withdrawn_or_archived` state and scope methods with `withdrawn`
* Remove any remaining references to the "archived" state

Trello: https://trello.com/c/nhvXmcY9/127-changing-archiving-to-withdrawing-in-whitehall-and-on-frontend

### CODE REVIEW

File: app/helpers/admin/editions_helper.rb [hunk 1]
```diff
@@ -279,8 +283,8 @@ def translation_preview_links(edition)
     end
   end
 
-  def archiving_or_unpublishing(edition)
-    edition.unpublishing.unpublishing_reason_id == UnpublishingReason::Archived.id ? 'archiving' : 'unpublishing'
+  def withdrawal_or_unpublishing(edition)
```
Comment (tommyp) [inline app/helpers/admin/editions_helper.rb]: Should this be withdrawing?

File: app/helpers/admin/editions_helper.rb [hunk 2]
```diff
@@ -279,8 +283,8 @@ def translation_preview_links(edition)
...
+    edition.unpublishing.unpublishing_reason_id == UnpublishingReason::Withdrawn.id ? 'withdrawal' : 'unpublishing'
```
Comment (tommyp) [inline app/helpers/admin/editions_helper.rb]: Same as above.

**Comment by tommyp** [comment]
:+1:

File: app/helpers/admin/editions_helper.rb (hunk 2 above)
Comment (tekin) [inline app/helpers/admin/editions_helper.rb]: I think it's 6 of one, half dozen of the other. "Edit public explanation for withdrawal" vs "Edit public explanation for withdrawing". Maybe a product decision? @bmwelby?

Comment (tommyp) [inline app/helpers/admin/editions_helper.rb]: I was going by the language used for `Archiving`. Either/or works. It's only minor.

Comment (tekin) [inline app/helpers/admin/editions_helper.rb]: Yeah, my thinking was that we have a better option with "withdraw" than adding an "ing" onto the end, i.e. the thing that represents the outcome of withdrawing a document is a withdrawal. There isn't an equivalent noun for unpublishing or archiving.

**Pull request merged**